### PR TITLE
Add ability to use User ID for LunaSea notifications

### PR DESCRIPTION
**Pull request by bobby-ore**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description
Add the ability to use User ID for LunaSea notifications (Keeps all existing functionality).

## Type of change

Please delete any options that are not relevant.

- User interface (UI)
- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

Please do not use any external image service. Instead, just paste in or drag and drop the image here, and it will be uploaded automatically.

### CODE REVIEW

File: server/notification-providers/lunasea.js [hunk 1]
```diff
@@ -8,15 +8,20 @@ class LunaSea extends NotificationProvider {
 
     async send(notification, msg, monitorJSON = null, heartbeatJSON = null) {
         let okMsg = "Sent Successfully.";
-        let lunaseadevice = "https://notify.lunasea.app/v1/custom/device/" + notification.lunaseaDevice;
+        let lunaseaurl = "";
+        if (notification.lunaseaNotificationType === "device") {
```
[the remark this answers is not in the document]
Comment (louislam) [reply server/notification-providers/lunasea.js:12]: I think you just need to swap the if-else

```
if (type === user) {
   // user url
} else {   // undefined type should go into this
   // same as previous
}
```

[the remark this answers is not in the document]
Comment (bobby-ore) [reply server/notification-providers/lunasea.js]: I swapped the if-else and changed 
`<input id="lunasea-device" v-model="$parent.notification.lunaseaId" type="text" class="form-control" required>`
back to
`<input id="lunasea-device" v-model="$parent.notification.lunaseaDevice" type="text" class="form-control" required>`
so there would not be any breaking changes with the data being saved and referenced in the notification provider.

[the remark this answers is not in the document]
Comment (bobby-ore) [reply server/notification-providers/lunasea.js]: I swapped the if-else and changed 
`<input id="lunasea-device" v-model="$parent.notification.lunaseaId" type="text" class="form-control" required>`
back to
`<input id="lunasea-device" v-model="$parent.notification.lunaseaDevice" type="text" class="form-control" required>`
so there would not be any breaking changes with the data being saved and referenced in the notification provider.

**Comment by louislam** [comment]
1. The `Device ID or User ID` is empty for old setup notifications.

image


2. Although it is functional, but I feel those naming could be confusing for future development

For examples: 

- `<label for="lunasea-device" class="form-label">{{ $t("LunaSea ID") }}`, LunaSea ID is representing user id and device id, but it called `lunasea-device` in code
- Is `LunaSea ID` really a good word? I just have a quick look of their doc, it seems that they never called it `LunaSea ID`


cc: @NiNiyas

**Comment by bobby-ore** [comment]
I’m sure that we can default the dropdown to use “Device ID”

I do agree that using `lunasea-device` for both user id and device id doesn’t make sense but then we will incur a breaking change. If we want to incur a breaking change I can update it to be named more appropriately. 

I chose `LunaSea ID` to represent the unique id given by LunaSea for the specific User or specific Device when grabbing the entire unique URL from the application, but we can change that naming if you would prefer.

**Comment by louislam** [comment]
I think a better design would be:

1. Dropdown: 
    - Name: `Target`
    - Options `device` (Default), `user`
    - Field Name: `notification.lunaseaTarget`
3. Input: 
    - Display Name: `Device ID`
    - Field Name: `notification.lunaseaDevice`
    - Show only if `lunaseaTarget` == `device`
5. Input: 
    - Display Name: `User ID`
    - Field Name: `notification.lunaseaUserID`
    - Show only if `lunaseaTarget` == `user`

In this way, there will be no breaking changes and the variable names is more clear.

**Comment by bobby-ore** [comment]
Preface, I am new to VueJs.

I'm having issues working with the `Target` Dropdown since the v-model for that is `v-model="$parent.notification.lunaseaTarget"` which if it is not populated (null) it will automatically create an empty option. I am not able to default the option to `device` since I have no way to reference that object since the `v-model` is null. Do you have any suggestions? Once I can get that figured out the rest should be straight forward for me.

**Comment by louislam** [comment]
You can add a script at the bottom and check it in the `mounted()`

Reference: https://github.com/louislam/uptime-kuma/blob/master/src/components/notifications/Gotify.vue

**Pull request merged**